app/controllers/friendship_controller.rb

```diff
@@ -0,0 +1,40 @@
+class FriendshipController < ApplicationController
+  before_action :authenticate_user!
+  before_action :set_friend, only: %i[destroy update]
+
+  def req_init
+    set_friendship = Friendship.where(user_id: params[:id], friend_id: current_user.id)
+    if set_friendship.exists?
+      set_friendship.update_all(confirmed: true)
+      redirect_to users_path
+      return
+    end
+    new_friendship = Friendship.where(user_id: current_user.id, friend_id: params[:id]).exists?
+    if !new_friendship
+      friend_obj = Friendship.new(user_id: current_user.id, friend_id: params[:id], confirmed: false)
+      friend_obj.save
+      flash[:success] = 'Friend request sent'
+    else
+      friend_obj = Friendship.where(user_id: current_user.id, friend_id: params[:id]).select('id')
+      Friendship.destroy(friend_obj.ids)
+      flash[:success] = 'Friend request cancelled'
+    end
+    redirect_to users_path
+  end
```

Comment:
This method is quit long. Can we split it into several methods, each having a single responsibility. Like we can have `create` method for only creating a friendship record, `destroy` for cancelling a friend request, `update` for accepting a friend request.
You can read more about single responsibility principle [here](https://www.sitepoint.com/solid-ruby-single-responsibility-principle/)